src/framework/audio/internal/midiplayer.cpp

```diff
@@ -137,31 +140,41 @@ void Sequencer::process(float sec, Context* ctx)
     tick_t toTick = curTick;
     if (toTick > maxValidTick) {
         toTick = maxValidTick;
-        if (ctx) {
-            ctx->playTick = prevTicks;
-            ctx->fromTick = prevTicks;
-            ctx->toTick = toTick;
-        }
+    }
 
-        if (m_streamState.requested) {
-            return;
-        }
+    while (m_streamState.requested) {
+        //wait NotationPlayback send data
     }

```

Comment:
We cannot block the entire thread.
If we do as intended so that the sequencer works in only one thread (for example, in Web Worker case), then this code will block the thread and, accordingly, block the receiving of messages, and accordingly, the data will never be received and the value of this flag will never change. Deadlock occurs.